hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,387 @@
+        command = 'interfaces switchport %s %s' % \
+            (self.interface_type, interface)
+        payload = self._make_payload(SHOW, command)
+        response = self._make_request('POST', url, data=payload)
+        # find the Native Vlan ID from the response
+        response = response.text.replace(' ', '')
+        if response.find('NativeVlanId:') == -1:
+            return None
+        begin = response.find('NativeVlanId:') + len('NativeVlanId:')
+        end = response.find('.', begin)
+        vlan = response[begin:end]
+        # a really bad temporary workaround to get the deployment tests to pass
+        if vlan == '1':
+            return None
+        return ('vlan/native', vlan)
+
+    def _add_vlan_to_trunk(self, interface, vlan):
+        """ Add a vlan to a trunk port.
+
+        If the port is not trunked, its mode will be set to trunk.
+
+        Args:
+            interface: interface to add the vlan to
+            vlan: vlan to add
+        """
+        if not self._is_port_on(interface):
+            self._port_on(interface)
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n tagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _remove_vlan_from_trunk(self, interface, vlan):
+        """ Remove a vlan from a trunk port.
+
+        Args:
+            interface: interface to remove the vlan from
+            vlan: vlan to remove
+        """
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n no tagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _remove_all_vlans_from_trunk(self, interface):
+        """ Remove all vlan from a trunk port.
+
+        Args:
+            interface: interface to remove the vlan from
+
+        this is kind slow for a number of reasons
+        1. we are using the REST API CLI
+        2. we have to go to every vlan and then remove our port from it;
+           that makes multiple calls.
+        """
+        for vlan in self._get_vlans(interface):
+            self._remove_vlan_from_trunk(interface, vlan[1])
+
+    def _set_native_vlan(self, interface, vlan):
+        """ Set the native vlan of an interface.
+
+        Args:
+            interface: interface to set the native vlan to
+            vlan: vlan to set as the native vlan
+
+        Method relies on the REST API CLI which is slow
+        """
+        if not self._is_port_on(interface):
+            self._port_on(interface)
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n untagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
```

Comment:
This should probably get factored out into a helper method; it looks like almost the same copypasta from _get_vlans() and _get_native_vlan().